sdv/multi_table/hma.py

```diff
@@ -44,6 +44,88 @@ def __init__(self, metadata, locales=None, verbose=True):
             self._table_synthesizers,
             self._table_sizes)
 
+    def _num_extended_columns(self, table_name, parent_name, columns_per_table):
+        """Get the number of columns that will be generated for table_name.
+
+        A table generates:
+            - 1 num_rows column for each for each foreign key with a specific parent
+            - n*(n-1)/2 correlation columns for each data column
+            - 4 parameters columns for each data column, with:
+                - 1 column for parameter a
+                - 1 column for parameter b
+                - 1 column for parameter scale
+                - 1 column for parameter loc
+        """
+        # The `num_rows` column. Because HMA models at most one relationship between
+        # two tables, this is always 1
+        num_cardinality_columns = 1
+
+        # no parameter columns are generated if there are no data columns
+        num_data_columns = columns_per_table[table_name]
+        if num_data_columns == 0:
+            return num_cardinality_columns
+
+        num_correlation_columns = (num_data_columns - 1) * num_data_columns // 2
+        num_parameters_columns = num_data_columns * 4
+
+        return num_correlation_columns + num_cardinality_columns + num_parameters_columns
+
+    def _estimate_columns_traversal(self, table_name, columns_per_table, visited):
+        """Given a table, estimate how many columns each parent will model.
+
```

Comment:
can you add the parameters here. It will be helpful since this method modifies objects in place instead of returning something at the end